insights/insightsuploader/insightsuploader.go

```diff
@@ -75,110 +102,60 @@ func (c *Controller) Run(ctx context.Context) {
 
 	// the controller periodically uploads results to the remote insights endpoint
 	cfg := c.configurator.Config()
-	configCh, cancelFn := c.configurator.ConfigChanged()
+	_, cancelFn := c.configurator.ConfigChanged()
 	defer cancelFn()
 
 	enabled := cfg.Report
 	endpoint := cfg.Endpoint
-	interval := cfg.Interval
-	initialDelay := wait.Jitter(interval/8, 2)
 	lastReported := c.reporter.LastReportedTime()
-	if !lastReported.IsZero() {
-		next := lastReported.Add(interval)
-		if now := time.Now(); next.After(now) {
-			initialDelay = wait.Jitter(now.Sub(next), 1.2)
-		}
-	}
-	if c.reporter.SafeInitialStart() {
-		initialDelay = 0
-	}
-	klog.V(2).Infof("Reporting status periodically to %s every %s, starting in %s", cfg.Endpoint, interval, initialDelay.Truncate(time.Second))
 
 	wait.Until(func() {
-		if initialDelay > 0 {
-			select {
-			case <-ctx.Done():
-			case <-time.After(initialDelay):
-			case <-configCh:
-				newCfg := c.configurator.Config()
-				interval = newCfg.Interval
-				endpoint = newCfg.Endpoint
-				if newCfg.Report != enabled {
-					enabled = newCfg.Report
-					if !newCfg.Report {
-						klog.V(2).Infof("Reporting was disabled")
-						initialDelay = newCfg.Interval
-						return
-					}
-					klog.V(2).Infof("Reporting was enabled")
-				}
-			}
-			initialDelay = 0
-		}
-
 		// attempt to get a summary to send to the server
 		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 		defer cancel()
 
-		source, ok, err := c.summarizer.Summary(ctx, lastReported)
-		if err != nil {
-			c.Simple.UpdateStatus(controllerstatus.Summary{Reason: "SummaryFailed", Message: fmt.Sprintf("Unable to retrieve local insights data: %v", err)})
-			return
-		}
-		if !ok {
-			klog.V(4).Infof("Nothing to report since %s", lastReported.Format(time.RFC3339))
-			return
-		}
-		defer source.Close()
-
-		if enabled && len(endpoint) > 0 {
-			// send the results
+		if len(endpoint) > 0 {
 			start := time.Now()
-			id := start.Format(time.RFC3339)
-			klog.V(4).Infof("Uploading latest report since %s", lastReported.Format(time.RFC3339))
-			if err := c.client.Send(ctx, endpoint, insightsclient.Source{
-				ID:       id,
-				Type:     c.client.GetMimeType(),
-				Contents: source,
-			}); err != nil {
-				klog.V(2).Infof("Unable to upload report after %s: %v", time.Now().Sub(start).Truncate(time.Second/100), err)
-				versionError := err == insightsclient.ErrWaitingForVersion || err == insightsclient.ErrObtainingForVersion
-				if versionError {
-					initialDelay = wait.Jitter(interval/8, 1) - interval/8
-					if c.reporter.SafeInitialStart() {
-						initialDelay = wait.Jitter(time.Second*15, 1)
+			sourceIndex := 0
```

Comment:
What is the intention of this please?

Reply:
Code currently only supports sending the "latest" package in a directory. However if the contents is too large it needs to be split across multiple payloads. This allows for all items in the directory to be processed.

Reply:
OK thanks